tests/test_config.py

```
@@ -70,78 +71,173 @@ def test_env_config_with_setup(self):
-        config = EnvConfig(
-            exports={"HOME_REF": "$HOME", "NESTED": "/scratch/${USER}/out"}
-        )
-        cmds = config.get_setup_commands()
-        assert 'export HOME_REF="$HOME"' in cmds
-        assert 'export NESTED="/scratch/${USER}/out"' in cmds
-
-    def test_exports_rejects_invalid_key(self):
-        config = EnvConfig(exports={"FOO;BAR": "value"})
+    def test_export_rejects_invalid_key(self):
+        config = EnvConfig(setup=[{"export": {"FOO;BAR": "value"}}])
         with pytest.raises(ValueError):
```

Ruff (0.15.17)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)